src/query/api/v1/handler/prometheus/native/common_test.go

```diff
@@ -379,37 +384,61 @@ func TestRenderInstantaneousResultsJSON(t *testing.T) {
 		Meta:   block.NewResultMetadata(),
 	}
 
-	renderResultsInstantaneousJSON(buffer, readResult)
-	expected := xtest.MustPrettyJSONMap(t, xjson.Map{
+	foo := xjson.Map{
+		"metric": xjson.Map{
+			"bar": "baz",
+			"qux": "qaz",
+		},
+		"value": xjson.Array{
+			1535948880,
+			"1",
+		},
+	}
+
+	bar := xjson.Map{
+		"metric": xjson.Map{
+			"baz": "bar",
+			"qaz": "qux",
+		},
+		"value": xjson.Array{
+			1535948880,
+			"2",
+		},
+	}
+
+	nan := xjson.Map{
+		"metric": xjson.Map{
+			"baz": "bar",
+		},
+		"value": xjson.Array{
+			1535948880,
+			"NaN",
+		},
+	}
+
+	buffer := bytes.NewBuffer(nil)
+	renderResultsInstantaneousJSON(buffer, readResult, true)
+	expectedWithNaN := xtest.MustPrettyJSONMap(t, xjson.Map{
 		"status": "success",
 		"data": xjson.Map{
 			"resultType": "vector",
-			"result": xjson.Array{
-				xjson.Map{
-					"metric": xjson.Map{
-						"bar": "baz",
-						"qux": "qaz",
-					},
-					"value": xjson.Array{
-						1535948880,
-						"1",
-					},
-				},
-				xjson.Map{
-					"metric": xjson.Map{
-						"baz": "bar",
-						"qaz": "qux",
-					},
-					"value": xjson.Array{
-						1535948880,
-						"2",
-					},
-				},
-			},
+			"result": xjson.Array{foo, nan, bar},
 		},
 	})
-	actual := xtest.MustPrettyJSONString(t, buffer.String())
-	assert.Equal(t, expected, actual, xtest.Diff(expected, actual))
+	actualWithNaN := xtest.MustPrettyJSONString(t, buffer.String())
+	assert.Equal(t, expectedWithNaN, actualWithNaN, xtest.Diff(expectedWithNaN, actualWithNaN))
+
+	buffer = bytes.NewBuffer(nil)
+	renderResultsInstantaneousJSON(buffer, readResult, false)
+	expectedWithoutNaN := xtest.MustPrettyJSONMap(t, xjson.Map{
+		"status": "success",
+		"data": xjson.Map{
+			"resultType": "vector",
+			"result": xjson.Array{foo, bar},
+		},
+	})
+	actualWithoutNaN := xtest.MustPrettyJSONString(t, buffer.String())
+	assert.Equal(t, expectedWithoutNaN, actualWithoutNaN, xtest.Diff(expectedWithoutNaN, actualWithoutNaN))
```

Comment:
Could be worth adding a test with just NaNs

Reply:
Added NaNs-only test case.